Aptoide/aptoide-client-v8
# [Feature/ASV-1905] Support Splits

**What does this PR do?**

   Shows Apps with splits in Search results, App view, Apps view, every list of apps (bundles, recommended).
   Potentially shows Apps with splits in Editorial content, Promotions (these need to be supported in the API side first).
   Adapted DownloadFactory to create Download's with base.apk+splits. This was where most of the work was done in order to allow splits download.
   Adapted AptoideDownloadManager to support split downloads (small changes to some mappers).
   Adapted InstallManager to support split installations and redirected the installs to the AppInstaller, which uses PackageInstaller to install apps with splits.
   Fixed an issue with moveCompletedDownloadFiles where it wasn't correctly moving apks and obbs after the first app was downloaded and correctly moved there.
   Database objects like Download / FileToDownload and Updates had to be changed to allow splits metadata.

   File system logic is now the following:
- Everything being downloaded is temporarly under .aptoide/ dir (this remains unchanged)
- Splits have the following name pattern: split-md5+split-name+.apk
- After App with splits is downloaded, splits are moved under: .aptoide/<app_name>-splits/

An app with splits is only valid if it respects the 
`"required_split_types": [
        "ABI",
        "DENSITY"
      ]`
Json in all aab responses.
The app is valid if ws response contains at least a split for each required split type. In this example the splits array would have to have at least an ABI and a DENSITY split to be considered valid. At the moment if it is not valid (it shouldn't happen since it should be correctly filtered in API) the download does not start.

In the realm migration I had to use a hack to have a RealmList<String> for the required splits. below realm 4.0.0 you cannot have a RealmList of primitive types, so I had to use an object (RealmString.java) with only a primitive type in it: String. The class already existed in the project and it was used somewhere in the class, that is why I check in the migration if the schema for that class already exists... Hopefully, this will all go away when we ditch realm.

  Installing state in app view was centered and is now larger, so it doesn't show that weird off-centered indeterminate state while installing. Also no progress is shown while installing.
  
  Some of the changes were Codestyle changes... I hope it's not a linux vs macos codestyle problem.

**Database changed?**

   Yes

**Where should the reviewer start?**

- [ ] v7.java (this will be all the "enable" aab's in requests changes
- [ ] DownloadFactory.java (here you will have almost everything related to the download of splits logic + naming and file paths)
- [ ] All app network objects
- [ ] All view models and models that have app information
- [ ] DefaultInstaller.java
- [ ] RealmToRealmDatabaseMigration.java

**How should this be manually tested?**

  There is a store with an app with splits. The store was created by FReis and it is called "split-store". Currently it has 1 app: linkedin (com.linkedin.android). This app is version 4.1.335.1 and its only uploaded in this store. There is no superior version than this in Aptoide.
This version currently only has 2 splits added (an armv8a abi split and a xxhdpi density split), which means that you will **NOT** see it if you don't have a compatible phone (needs to be compatible with armv8a, usually an armv7a phone will be compatible, but a x86 [emulators usually] won't). 
Test compatible device > app shows up in splits-store.
Test uncompatible device > app does not show up (# of applications in store still is 1... which is weird for the user and probably a bug in the api).
This linkedin app should show up in Search results, app view, apps (if you download an earlier version and follow split-store).

After everything about showing up aab's is tested, test:
- [ ] Download com.linkedin.android with splits without linkedin installed. (check .aptoide in file system to see if files are being handled correctly)
- [ ] Update to com.linked.android with splits in appview and apps (install an earlier linked in version first without splits)
- [ ] Test Downgrade, from splits to older version.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1905](https://aptoide.atlassian.net/browse/ASV-1905)
  Documentation related to this pull-request: [App Bundles API](https://aptoide.atlassian.net/wiki/spaces/AN/pages/1358725187/App+Bundle+api)


**Code Review Checklist**
